src/components/Wallet/ExtendedPublicKeySelector.jsx

```diff
@@ -81,10 +81,14 @@ class ExtendedPublicKeySelector extends React.Component {
   getAssociatedExtendedPublicKeyImporter = () => {
     const { signingKeys, number, extendedPublicKeyImporters } = this.props;
     const { selection } = this.state;
+    let associatedExtendedPublicKeyImporter = null;
 
-    const keyIndex = number > 0 ? signingKeys[number - 1] : selection;
-    if (keyIndex > 0) return extendedPublicKeyImporters[keyIndex];
-    return null;
+    if (selection) {
+      associatedExtendedPublicKeyImporter =
+        extendedPublicKeyImporters[selection];
+    }
+
+    return associatedExtendedPublicKeyImporter;
```

Comment:
Definitely could use some eyes on this one.

Reply:
This looks like it makes sense to me, but the prop `number` feels vague and so not sure if we're missing something there. Do you know where this is being used? I can help test functionality.

Reply:
It looks like it's being used when an xpub is selected for signing and also in the address ex pander to verify ownership (I think):

```
[walden:~/code/caravan] upgrade-packages+* ± grep -nr "<ExtendedPublicKeySelector" src/
src/components/Wallet/WalletSign.jsx:87:          <ExtendedPublicKeySelector number={keySelectorNum} />
src/components/Wallet/AddressExpander.jsx:264:        <ExtendedPublicKeySelector number={0} onChange={this.keySelected} />
```